src/WebJobs.Script.WebHost/Standby/StandbyManager.cs

```diff
@@ -107,12 +116,29 @@ public async Task InitializeAsync()
                     File.WriteAllText(Path.Combine(functionPath, "run.csx"), content);
 
                     _logger.LogInformation($"StandbyMode placeholder function directory created");
+
+                    // start a background timer to identify when specialization happens
+                    // specialization usually happens via an http request (e.g. scale controller
+                    // ping) but this timer is started as well to handle cases where we
+                    // might not receive a request
+                    _specializationTimer = new Timer(OnSpecializationTimerTick, null, _specializationTimerInterval, _specializationTimerInterval);
                 }
                 finally
                 {
                     _semaphore.Release();
                 }
             }
         }
+
+        private void OnSpecializationTimerTick(object state)
+        {
+            if (!_environment.InStandbyMode)
+            {
+                SpecializeHostAsync().GetAwaiter().GetResult();
```

Comment:
Do we need to block here? We probably don't want any failures here to throw on the timer callback anyway.

Reply:
Also with this as is... we'll have callbacks blocking here until specialization is done... if it takes 2 secs, we'd have 4 or them. Worth disabling the timer before specializing

Reply:
Our comments got crossed -- yeah, I'm going to change this. Good point about disabling the timer first. I'll do that and then fire-and-forget the specialization (with a log in a continuation if faulted).

Reply:
My v1 code disposed the timer first thing when specializing (code [here](https://github.com/Azure/azure-functions-host/blob/v1.x/src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs#L142)). Recommend reviewing that code as much as possible to make sure we're not missing any other things that we had before. It was hard to get it all right.